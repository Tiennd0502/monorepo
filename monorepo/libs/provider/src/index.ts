export { default as TamaguiProvider } from './TamaguiProvider';
export { default as QueryClientProvider } from './QueryClientProvider';
