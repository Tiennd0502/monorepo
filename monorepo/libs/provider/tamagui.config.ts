import { createTamagui, createTokens } from 'tamagui';
import { config } from '@tamagui/config/v3';
import { themes } from '@tamagui/themes';

import {
  colorsMapping,
  radius,
  size,
  space,
  zIndex,
  defaultFont,
  headingFont,
} from '@monorepo/ui';

const tokens = {
  radius,
  size,
  space,
  zIndex,
};

const tamaguiConfig = createTamagui({
  ...config,
  defaultFont: 'body',
  themes,
  fonts: {
    body: defaultFont,
    heading: headingFont,
  },
  tokens: createTokens({
    ...tokens,
    color: colorsMapping,
  }),
});

type AppConfig = typeof config;

declare module 'tamagui' {
  type TamaguiCustomConfig = AppConfig;
}

export default tamaguiConfig;
