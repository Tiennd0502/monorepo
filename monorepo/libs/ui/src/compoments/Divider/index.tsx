import { memo } from 'react';
import { Stack, StackProps } from 'tamagui';

export interface DividerProps extends StackProps {
  width?: string;
  height?: string;
  color?: string;
}

const Divider = ({
  width = '100%',
  height = '$px',
  color = '$backgroundDisabled',
  ...props
}: DividerProps) => (
  <Stack width={width} height={height} backgroundColor={color} {...props} />
);

export default memo(Divider);
