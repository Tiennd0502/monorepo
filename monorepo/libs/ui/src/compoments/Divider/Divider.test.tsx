import {render} from '@testing-library/react-native';

import { TamaguiProvider } from '@monorepo/provider';
import Divider from '.';

const baseProps = {};

const setup = (moreProps = {}) => {
  const props = {
    ...baseProps,
    ...moreProps,
  };
  
  return render(
    <TamaguiProvider>
      <Divider {...props} />
    </TamaguiProvider>
  );
};

describe('Divider', () => {
  it('should render Avatar component with default props', () => {
    const container = setup();

    expect(container).toMatchSnapshot();
  });

  it('should render Avatar component with props', () => {
    const container = setup({height: '2', color: 'red'});

    expect(container).toMatchSnapshot();
  });
});
