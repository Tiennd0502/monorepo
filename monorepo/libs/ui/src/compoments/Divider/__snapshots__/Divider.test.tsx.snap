// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider should render Avatar component with default props 1`] = `
<View
  style={
    {
      "backgroundColor": "#909090",
      "height": 1,
      "width": "100%",
    }
  }
/>
`;

exports[`Divider should render Avatar component with props 1`] = `
<View
  style={
    {
      "backgroundColor": "red",
      "height": "2",
      "width": "100%",
    }
  }
/>
`;
